jenkins/run-server-func-tests

```diff
@@ -8,9 +8,20 @@ export PB_SERVER_IMAGE_TAG=${PB_SERVER_IMAGE_TAG:-"$(cat jenkins/branch.name)"}
 export PB_POD_NAME=${PB_POD_NAME:-"pbench-in-a-can_${PB_SERVER_IMAGE_TAG}"}
 export PB_SERVER_CONTAINER_NAME=${PB_SERVER_CONTAINER_NAME:-"${PB_POD_NAME}-pbenchserver"}
 
-SERVER_URL="http://localhost:8080"
+# Note:  the value of PB_HOST_IP will be used to generate the TLS certificate
+# and so it (not `localhost`) must also be used to access the Pbench Server;
+# otherwise, the TLS validation will fail due to a host mismatch.
+if [[ -z "${PB_HOST_IP}" ]]; then
+    host_ip_list=$(hostname -I)
+    PB_HOST_IP=${host_ip_list%% *}
+    export PB_HOST_IP
+fi
+SERVER_URL="https://${PB_HOST_IP}:8443"
```

Comment:
Does this work on non-routable CSB addresses? 😦 

For example, mine shows a local router address first, and *then* the VPN IPv4, followed by a couple of IPv6 addresses. You're grabbing the local house IP, which won't be reachable from outside.

Then again, maybe it just means I always need to override `PB_HOST_IP`; I just wish we didn't have so many of these hidden environment configuration variables in different forms (e.g., `PB` vs `PBENCH`), or they were easier to discover. (Sigh.)

Reply:
> Does this work on non-routable CSB addresses?

I don't know...does it?  🤔 

My understanding of the concept here is that the Pbench Server is run on the same node as this script, so, as long as the access is coming from the node where this script is run (as would be the case with the functional tests...), everything should work.  However, you're correct that it might not work for accesses coming from other nodes (e.g., if this script is run on a VM in RDU and I try to hit the Server from my Mac here, it won't work if the first IP address is non-routable and/or if I use an address other than the one which this script chooses).

Unfortunately, I don't really know how to do better.  (Got any thoughts which aren't hacks?  😁)  The man page for `hostname -I` explicitly says that I cannot depend on the ordering of the returned addresses.  😢  So, yeah, you have the option of providing `PB_HOST_IP`....

I suppose that I could try to make the certificate generation include _all_ the addresses returned by `hostname -I`...that sounds...challenging, but it might solve the problem.  (But, I would prefer to defer that to another PR....)

> I just wish we didn't have so many of these hidden environment configuration variables in different forms (e.g., `PB` vs `PBENCH`)

I agree.  However, having too many is better than having too few (where you cannot make the tool work without hacking it...).  And, yes, I wish we had a uniform convention for naming them (but, the 6+ letter prefix proved awkwardly long...).

> they were easier to discover.

Yeah, I don't have much to offer on that.  I try to either document them at the top of the script in a big block comment (which people then try to tell me should be moved to a `README` file...), or I just reference them at the top of the script in an effort to make the code self-documenting, either way, hopefully, a quick look at the top of the script should reveal the options.

Reply:
This instance probably isn't a big deal since it's not intended for real deployment and should always be co-hosted and independent. For the other instance ... I'm less sure ... but probably we never do real deployment on a host with a non-routable IP, so maybe *that* doesn't matter, too?

Reply:
> The problem comes when there is a human user involved, because they often do things like run a browser or Postman, and they do it on their workstation after kicking off the canned Server somewhere else

Of course the whole point of getting a beefed-up laptop refresh (and then moving to Fedora 38 CSB) was so I can run my browser, Postman, and server all on the laptop. This is now almost entirely painless except for the `8080` conflict between tool meister and server when I add the Agent. 🌤️

Reply:
> This is now almost entirely painless except for the `8080` conflict between tool meister and server when I add the Agent.

Yeah...we should figure out how to fix that!  Maybe we should move the Server to some other port...like `8443`....  😉

Reply:
> Maybe we should move the Server to some other port...like `8443`

Ha; yeah, maybe someone will do that soon. 🥇 I'll have to try it out tomorrow...